nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -505,7 +505,8 @@ private void handleEqualityComparison(
     Nullness value = values(input).valueOfSubNode(node.getExpression());
     Node target = node.getTarget();
 
-    if (target instanceof LocalVariableNode) {
+    if (target instanceof LocalVariableNode
+        && !ASTHelpers.getType(target.getTree()).isPrimitive()) {
```

Comment:
How fast is `ASTHelpers.getType(...)`?

Reply:
It should be pretty cheap, as it basically just does a downcast and a field read:

https://github.com/google/error-prone/blob/fc250863985501f1d31aa8fc8215976ffab9e815/check_api/src/main/java/com/google/errorprone/util/ASTHelpers.java#L1079-L1082

But yes, I should run the perf tests to see if any win is observable; will do.

BTW, When we get NullAway running on NullAway I would add `castToNonNull` around these `getType()` calls, as they should never return `null` in these cases.